zorg/buildbot/builders/UnifiedTreeBuilder.py

```diff
@@ -567,3 +571,448 @@ def getCmakeWithNinjaMultistageBuildFactory(
+    cmake_definitions = cmake_definitions or dict()
+    assert isinstance(cmake_definitions, dict), "The CMake definitions argument must be a dictionary."
+    cmake_options = cmake_options or list()
+    assert isinstance(cmake_options, list), "The CMake options argument must be a list."
+
+    if not "LLVM_ENABLE_PROJECTS" in cmake_definitions and f.enable_projects:
+        cmake_definitions.update({ "LLVM_ENABLE_PROJECTS" : ";".join(f.enable_projects) })
+
+    if not "LLVM_ENABLE_RUNTIMES" in cmake_definitions and f.enable_runtimes:
+        cmake_definitions.update({ "LLVM_ENABLE_RUNTIMES" : ";".join(f.enable_runtimes) })
+
+    if not "CMAKE_INSTALL_PREFIX" in cmake_definitions and f.install_dir:
+        cmake_definitions.update({ "CMAKE_INSTALL_PREFIX" : LLVMBuildFactory.pathRelativeTo(
+                                                                f.install_dir,
+                                                                f.obj_dir) })
+        # Remove install directory.
+        f.addSteps([
+            steps.RemoveDirectory(
+                name            = f"clean-install-dir",
+                dir             = util.Interpolate(f.install_dir),
+                description     = ["Remove", util.Interpolate(f.install_dir), "directory"],
+                haltOnFailure   = False,
+                flunkOnFailure  = False,
+                doStepIf        = lambda step, clean = clean: clean or step.getProperty("clean_obj") == True
+            ),
+        ])
+
+    # Set proper defaults.
+    if allow_cmake_defaults:
+        if not "CMAKE_BUILD_TYPE" in cmake_definitions:
+            cmake_definitions.update({ "CMAKE_BUILD_TYPE" : "Release" })
+        if not "LLVM_ENABLE_ASSERTIONS" in cmake_definitions:
+            cmake_definitions.update({ "LLVM_ENABLE_ASSERTIONS" : "ON" })
+        if not "LLVM_LIT_ARGS" in cmake_definitions and checks:
+            cmake_definitions.update({ "LLVM_LIT_ARGS" : "-v -vv" })
+
+    f.addStep(
+        steps.CMake(
+            name            = f"cmake-configure",
+            path            = LLVMBuildFactory.pathRelativeTo(f.llvm_srcdir, f.obj_dir),
+            generator       = generator,
+            definitions     = cmake_definitions,
+            options         = cmake_options,
+            description     = ["CMake configure"],
+            haltOnFailure   = True,
+            env             = env,
+            workdir         = f.obj_dir
+        ))
+
+    # Build Commands.
+    if targets:
+        for target in targets:
+            cmake_build_options = ["--build", "."]
+            if target != ".":
+                cmake_build_options.extend(["--target", target])
+            if jobs:
+                cmake_build_options.extend(["--", "-j", jobs])
+
+            target_title = "default" if target == "." else target
+
+            f.addStep(
+                steps.CMake(
+                    name            = f"build-{target_title:.42}",
```

Comment:
We have the step name length increased to 255. See https://github.com/gkistanova/buildbot/commit/b745a335ebdbb58ec2f09b7e87242b72f2a3b0bc. No real need to truncate it to 42.

Reply:
The buildbot does not support the long step names by itself. The local copy of the buildbot will get unpredictable failed in case of specifying of the long step names I suppose.
Also, the long step names break the web ui. I think it is better to cut the step names to the max defaults.

Reply:
I'm fine with truncating the names, I'm not easy with hardcoded lengths, though. It will be a wrong length next time somebody will rename the step. But we can keep it as is if you feel strongly about that.

Reply:
Not a big deal actually. Now I think that's not my business to worry about the length of the step names. This is the buildbot's responsibility if there are any limits there.

Reply:
Exactly my point.